website/components/LoadModal.tsx

```diff
@@ -0,0 +1,162 @@
+
+  const modalContainerRef = useRef<HTMLDivElement>(null);
+
+  const setShowModal = (show: boolean): void => {
+    if (show) {
+      setUrlInput("");
+      setErrorText("");
+    }
+    _setShowModal(show);
+  };
+
+  const onClickLoad = (): void => {
+    // TODO: Handle multiple URLs?
+    // TODO: Do any transformation of URLs here? Currently just using the labels directly.
+    if (!isValidUrl(urlInput)) {
+      setErrorText("Please enter a valid URL.");
+      return;
+    }
+
+    const appProps: AppDataProps = {
+      imageUrl: urlInput,
+      imageDownloadHref: urlInput,
+      cellId: "1",
+      parentImageUrl: "",
+      parentImageDownloadHref: "",
+      // Enable first three channels by default
+      viewerChannelSettings: {
+        groups: [
+          {
+            name: "Channels",
+            channels: [
+              { match: [0, 1, 2], enabled: true },
+              { match: "(.+)", enabled: false },
+            ],
+          },
+        ],
+      },
+    };
+    props.onLoad(appProps);
+    addRecentDataUrl({ url: urlInput, label: urlInput });
+    setShowModal(false);
+  };
+
+  // Set up fuse for fuzzy searching on the labels of recent datasets
+  const fuse = useMemo(() => {
+    return new Fuse(recentDataUrls, {
+      keys: ["label"],
+      isCaseSensitive: false,
+      shouldSort: true, // sorts by match score
+      ignoreLocation: true, // search more than first 60 characters
+      findAllMatches: true, // return all matches
+    });
+  }, [recentDataUrls]);
+
+  // This search could be done using a transition if needed, but since there is a max of 100 urls,
+  // performance hits should be minimal.
+  const autoCompleteOptions: { label: React.ReactNode; value: string }[] = useMemo(() => {
+    let filteredItems: RecentDataUrl[] = [];
+    if (urlInput === "") {
+      // Show first 20 recent data urls
+      filteredItems = recentDataUrls.slice(0, MAX_RECENT_URLS_TO_DISPLAY);
+    } else {
+      // Show first 20 search results
+      filteredItems = fuse
+        .search(urlInput)
+        .slice(0, MAX_RECENT_URLS_TO_DISPLAY)
+        .map((option) => option.item);
+    }
+    return filteredItems.map((item) => {
+      return {
+        label: <MiddleTruncatedText text={item.label} />,
+        value: item.url,
+      };
+    });
+  }, [urlInput, fuse]);
+
+  const getAutoCompletePopupContainer = modalContainerRef.current ? () => modalContainerRef.current! : undefined;
+
+  return (
+    <ModalContainer ref={modalContainerRef}>
+      <Button type="link" onClick={() => setShowModal(!showModal)}>
+        <UploadOutlined />
+        Load
+      </Button>
+      <Modal
+        open={showModal}
+        title={"Load"}
+        onCancel={() => {
+          setShowModal(false);
+        }}
+        getContainer={modalContainerRef.current || undefined}
+        okButtonProps={{}}
+        footer={
+          <Button type="default" onClick={() => setShowModal(false)}>
```

Comment:
optional minor nit: two closures, both alike in content, one with a block and one inline